src/jit/emitarm.cpp

```diff
@@ -2459,6 +2459,16 @@ void emitter::emitIns_R_R_I(instruction ins,
                 fmt = IF_T2_M0;
                 sf  = INS_FLAGS_NOT_SET;
             }
+            else if (insDoesNotSetFlags(flags) && reg1 != REG_SP && reg1 != REG_PC)
+            {
+                // movw,movt reg1, imm
+                codeGen->instGen_Set_Reg_To_Imm(attr, reg1, imm);
+
+                // ins reg1, reg2
+                emitIns_R_R(ins, attr, reg1, reg2);
```

Comment:
Fixed, this was incorrect for `SUB`.

This is needed in legacy jit during generation of virtual method invocation at vtable-1st-level-indirection:

```cpp
if (isRelative)
{
#if defined(_TARGET_ARM_)
  unsigned offset = vtabOffsOfIndirection + vtabOffsAfterIndirection;

  // ADD vptrReg1, REG_CALL_IND_SCRATCH, vtabOffsOfIndirection + vtabOffsAfterIndirection
  getEmitter()->emitIns_R_R_I(INS_add, EA_PTRSIZE, vptrReg1, vptrReg, offset);
#else
  unreached();
#endif
}

// MOV vptrReg, [REG_CALL_IND_SCRATCH + vtabOffsOfIndirection]
getEmitter()->emitIns_R_AR(ins_Load(TYP_I_IMPL), EA_PTRSIZE, vptrReg, vptrReg, vtabOffsOfIndirection);
```

Immediate in `ADD` instruction may not be enough for `offset`. New case allows to `ADD/SUB` immediate without using additional register.